samples/vmc/sddc/deploy_ovf_template_cloud.py

```diff
@@ -0,0 +1,140 @@
+* WARRANTIES OR CONDITIONS OF MERCHANTABILITY, SATISFACTORY QUALITY,
+* NON-INFRINGEMENT AND FITNESS FOR A PARTICULAR PURPOSE.
+"""
+
+__author__ = 'VMware, Inc.'
+__vcenter_version__ = 'VMware Cloud on AWS'
+
+from com.vmware.content.library_client import Item
+from com.vmware.vcenter.ovf_client import LibraryItem
+from com.vmware.vcenter_client import ResourcePool
+from vmware.vapi.vsphere.client import create_vsphere_client
+
+from samples.vsphere.common import sample_cli, sample_util
+from samples.vsphere.common.id_generator import generate_random_uuid
+from samples.vsphere.contentlibrary.lib.cls_api_helper import ClsApiHelper
+
+
+class DeployOvfTemplateCloud:
+    """
+    Demonstrates the workflow to deploy an OVF library item to
+    a resource pool in VMware Cloud on AWS.
+    Note: the sample needs an existing library item with an OVF template
+    and an existing resource pool with resources for deploying the VM.
+    """
+
+    def __init__(self):
+        parser = sample_cli.build_arg_parser()
+
+        parser.add_argument('--resource_pool_name',
+                            default='Compute-ResourcePool',
+                            help='The name of the resource pool to be used.')
+
+        parser.add_argument('--libitemname',
+                            required=True,
+                            help='The name of the library item to deploy.'
+                                 'The library item should contain an OVF package.')
+
+        args = sample_util.process_cli_args(parser.parse_args())
+
+        self.vm_id = None
+        self.vm_name = 'deployed-vm-' + str(generate_random_uuid())
+
+        self.lib_item_name = args.libitemname
+        self.resource_pool_name = args.resource_pool_name
+        self.cleardata = args.cleardata
+
+        # Connect to vAPI Endpoint on vCenter Server
+        self.client = create_vsphere_client(server=args.server,
+                                            username=args.username,
+                                            password=args.password)
+
+        self.helper = ClsApiHelper(self.client, args.skipverification)
+
+    def deploy_ovf_template(self):
+
+        # Build the deployment target with resource pool ID
+        filter_spec = ResourcePool.FilterSpec(names=set([self.resource_pool_name]))
+        resource_pool_summaries = self.client.vcenter.ResourcePool.list(filter_spec)
+        if not resource_pool_summaries:
+            raise ValueError("Resource pool with name '{}' not found".
+                             format(self.resource_pool_name))
+        resource_pool_id = resource_pool_summaries[0].resource_pool
+        print("Resource pool ID: {}".format(resource_pool_id))
+        deployment_target = LibraryItem.DeploymentTarget(
+            resource_pool_id=resource_pool_id)
+
+        # Find the library item
+        find_spec = Item.FindSpec(name=self.lib_item_name)
+        lib_item_ids = self.client.content.library.Item.find(find_spec)
```

Comment:
Can we use ClsApiHelper.get_item_id_by_name instead? The self.helper is defined above but not used.